utils/spacewalk-common-channels.ini

```diff
@@ -548,15 +561,105 @@ checksum = sha256
 yumrepo_url = http://download.opensuse.org/update/leap/15.0/non-oss/
 
-[spacewalk28-client-opensuse_leap_15_0]
-name     = Spacewalk Client 2.8 %(base_channel_name)s
+[opensuse_leap15_0-uyuni-client]
+name     = Uyuni Client Tools for %(base_channel_name)s
 archs    = x86_64
 base_channels = opensuse_leap15_0-%(arch)s
 checksum = sha256
-gpgkey_url = http://download.opensuse.org/repositories/systemsmanagement:/spacewalk:/2.8/openSUSE_Leap_15.0/repodata/repomd.xml.key
-gpgkey_id = A0E46E11
-gpgkey_fingerprint = 68D3 3874 9967 0AEB D988  2DB3 2ABF A143 A0E4 6E11
-yumrepo_url = http://download.opensuse.org/repositories/systemsmanagement:/spacewalk:/2.8/openSUSE_Leap_15.0/
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_client_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_client_gpgkey_fingerprint)s
+yumrepo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/
+
+[opensuse_leap15_0-uyuni-client-devel]
+name     = Uyuni Client Tools for %(base_channel_name)s (Development)
+archs    = x86_64
+base_channels = opensuse_leap15_0-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_client_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_client_gpgkey_fingerprint)s
+yumrepo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/
+
+[sle12-sp3-uyuni-client]
+name     = Uyuni Client Tools for SLE12 SP3
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sles12-sp3-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_client_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_client_gpgkey_fingerprint)s
+yumrepo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE12-Uyuni-Client-Tools/SLE_12/
+
+[sle12-sp3-uyuni-client-devel]
+name     = Uyuni Client Tools for SLE12 SP3 (Development)
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sles12-sp3-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE12-Uyuni-Client-Tools/SLE_12/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_client_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_client_gpgkey_fingerprint)s
+yumrepo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE12-Uyuni-Client-Tools/SLE_12/
+
+[sle12-sp4-uyuni-client]
+name     = Uyuni Client Tools for SLE12 SP4
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sles12-sp4-pool-%(arch)s
```

Comment:
Now that I notice, I used this channel, but I guess this means this will NOT work if only SLED channel is present.

Am I right?

Is there any way we can make it work for both SLES and SLED?

Reply:
Cloning is the only option or they specify a new repo manually. If SLED would be the only problem I would define a special section, but we have also SLES for SAP and SLES for HPC.

I think we are good enough. The people who want to use uyuni with SLES have to do a bit more work.

Reply:
Since I think the regular use case is going to be plan SLES, I will create channels for that.

Then if someone wants SLED, SLES for SAP, SLES for HPC or whatever other variants, they can easily adapt this to do it and submit a PR :-)